bitwarden_access_manager.py

```diff
@@ -0,0 +1,672 @@
+        return self.id
+
+    @staticmethod
+    def from_toml_dict(data: Dict[str, Any]) -> Group:
+        access_all: Optional[bool] = False
+        if "access_all" in data:
+            access_all = data["access_all"]
+
+        return Group(
+            id=data["group_id"],
+            name=data["group_name"],
+            access_all = access_all,
+        )
+
+    def format_toml(self) -> str:
+        lines = [
+            "[[group]]",
+            f"group_id = {self.id}",
+            f"group_name = {self.name}",
+            f"access_all = {str(self.access_all).lower()}",
+        ]
+        return "\n".join(lines)
+
+
+class MemberCollectionAccess(NamedTuple):
+    name: str
+
+    def get_id(self) -> str:
+        return self.id
+
+    @staticmethod
+    def from_toml_dict(data: Dict[str, Any]) -> MemberCollectionAccess:
+        return MemberCollectionAccess(
+            name=data["member_name"],
+        )
+
+    def format_toml(self) -> str:
+        return (
+            "{ member_name = "
+            + self.name
+            + '"}'
+        )
+
+class GroupCollectionAccess(NamedTuple):
+    name: str
+    access: str
+
+    def get_id(self) -> str:
+        return self.id
+
+    @staticmethod
+    def from_toml_dict(data: Dict[str, Any]) -> GroupCollectionAccess:
+        return GroupCollectionAccess(
+            name=data["group_name"],
+            access=data["access"],
+        )
+
+    def format_toml(self) -> str:
+        return (
+            "{ group_name = "
+            + self.name
+            + '", access = "'
+            + self.access
+            + '" }'
+        )
+
+class Collection(NamedTuple):
+    id: str
+    external_id: str
+    group_access: Optional[Tuple[GroupCollectionAccess, ...]]
+    member_access: Optional[Tuple[MemberCollectionAccess, ...]]
+
+    def get_id(self) -> str:
+        return self.id
+
+    @staticmethod
+    def from_toml_dict(data: Dict[str, Any]) -> Collection:
+        group_access: Optional[Tuple[GroupCollectionAccess, ...]] = None
+        if "group_access" in data:
+            group_access = tuple(
+                sorted(
+                    GroupCollectionAccess.from_toml_dict(x) for x in data["group_access"]
+                )
+            )
+
+        member_access: Optional[Tuple[MemberCollectionAccess, ...]] = None
+        if "member_access" in data:
+            member_access = tuple(
+                sorted(
+                    MemberCollectionAccess.from_toml_dict(x) for x in data["member_access"]
+                )
+            )
+
+        return Collection(
+            id=data["collection_id"],
+            external_id=data["external_id"],
+            group_access=group_access,
+            member_access=member_access,
+      )
+
+    def format_toml(self) -> str:
+        result = (
+            "[[collection]]\n"
+            f"collection_id = {self.id}\n"
+            # Splicing the string is safe here, because Bitwarden repo names are
```

Comment:
This looks like a copy-paste bug, Bitwarden has collections, not repositories.

Also, I am not so sure about whether the comment is true ... on GitHub you cannot have `"` or `\` in a repository name, but collection names in Bitwarden are more relaxed, is could they not contain `"` or `\`?

Reply:
You're right, I tested and this comment is not true. A collection name or externalID can contain `"` or `\`: [7250ebf2](https://github.com/ChorusOne/github-access-manager/pull/10/commits/7250ebf24bb9894fcd2f2d262ddd92ad7c43b20f)